tests/models/falcon/test_modeling_falcon.py

```diff
@@ -510,6 +512,40 @@ def test_model_rope_scaling(self):
             torch.testing.assert_close(ntk_sin_long, original_sin_long)
         self.assertTrue((ntk_scaling_rope.inv_freq <= original_rope.inv_freq).all())
 
+        # Sanity check YaRN RoPE scaling
+        yarn_scaling_rope = FalconYaRNScalingRotaryEmbedding(
+            head_dim,
+            max_position_embeddings=config.max_position_embeddings,
+            base=config.rope_theta,
+            scaling_factor=scaling_factor,
+        ).to(torch_device)
+        yarn_cos_short, yarn_sin_short = yarn_scaling_rope(x, short_input_length)
+        yarn_cos_long, yarn_sin_long = yarn_scaling_rope(x, long_input_length)
+        torch.testing.assert_close(yarn_cos_short, yarn_cos_long[:short_input_length, :])
+        torch.testing.assert_close(yarn_sin_short, yarn_sin_long[:short_input_length, :])
+        with self.assertRaises(AssertionError):
+            torch.testing.assert_close(yarn_cos_long, original_cos_long)
+        with self.assertRaises(AssertionError):
+            torch.testing.assert_close(yarn_sin_long, original_sin_long)
```

Comment:
let's also check that yarn_sin/cos_short != original_sin/cos_short (i.e. that applying yarn should change all values)